lib/net.c

```diff
@@ -1776,9 +1776,220 @@ start_network_stage2(void *arg)
 }
 
 static int
-drop_privileges(void)
+copy_amb_to_inh(const char *name, cap_t cap_p)
+{
+	cap_value_t i;
+
+	for (i = 0; i <= CAP_LAST_CAP; i++) {
+		char *cap_name;
+		int ret;
+
+		int value = cap_get_ambient(i);
+		if (value < 0) {
+			cap_name = cap_to_name(i);
+			if (cap_name == NULL) {
+				G_LOG(WARNING, "%s (%s): could not get string for capability %u while reporting that it is not supported by the running kernel: %s\n",
+					__func__, name, i, strerror(errno));
+				continue;
+			}
+
+			G_LOG(WARNING, "%s (%s): capability %s (%u) not supported by the running kernel: %s\n",
+				__func__, name, cap_name, i, strerror(errno));
+			cap_free(cap_name);
+			continue;
+		}
+
+		ret = cap_set_flag(cap_p, CAP_INHERITABLE, 1, &i,
+			value ? CAP_SET : CAP_CLEAR);
+		if (ret != 0) {
+			cap_name = cap_to_name(i);
+			if (cap_name == NULL) {
+				G_LOG(WARNING, "%s (%s): could not get string for capability %u while reporting that it could not be set to CAP_INHERITABLE: %s\n",
+					__func__, name, i, strerror(errno));
```

Comment:
`errno` is coming from `cap_to_name()` instead of `cap_set_flag()`. We need to save `errno` before calling `cap_to_name()`.